evap/evaluation/models_logging.py

```diff
@@ -341,28 +341,51 @@ def unlogged_fields(self):
         return ["id", "order"]
 
 
+def _get_m2m_field_name(model_class, sender):
+    return next(
+        (field.name for field in model_class._meta.many_to_many if getattr(model_class, field.name).through == sender),
+        None,
+    )
+
+
 @receiver(m2m_changed)
-def _m2m_changed(sender, instance, action, reverse, model, pk_set, **kwargs):  # pylint: disable=unused-argument
-    if reverse:
+def _m2m_changed(sender, instance, action, reverse, model, pk_set, **kwargs):  # noqa: PLR0912
+    field_name = _get_m2m_field_name(model if reverse else type(instance), sender)
```

Comment:
This is the only caller now, I don't think the function is a nice abstraction, my vote is to inline this again.